association_backend/association/domain/entities/stripe_entities.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from pydantic import validator
+from pydantic.main import BaseModel
+
+
+@dataclass
+class StripeCustomer:
+    id: str
+    email: str
+
+
+@dataclass
+class StripeCheckoutSession:
+    id: str
+    customer_id: Optional[str] = ""
+    subscription_id: Optional[str] = ""
+
+
+class StripeCheckoutSessionInput(BaseModel):
+    customer_id: Optional[str] = None
+    customer_email: Optional[str] = None
+
+    @validator("customer_email")
+    def check_customer_id_or_customer_email(cls, customer_email, values):
+        if not values.get("customer_id") and not customer_email:
+            raise ValueError("either customer_id or customer_email is required")
```

Comment:
nitpick: 

why do we offer two ways to do it? I wonder if we could refactor the code so we always either have the id or the email

Reply:
Case 1: Logged User bought a ticket for Pycon and he is already a registered Customer in Stripe -> we can create a CheckoutSession passing his stripe `customer_id`
Case 2: Otherwise -> you can create both CheckoutSession and Customer on Stripe passing `customer_email` key

I think this is the best way...

Reply:
We could use the ID for cases? or the email?

Reply:
If you use `customer_email`, a new Customer is created
If you don't use `customer_email`, you have to call the API `Customer.create` before calling these, I think it is not what we want
Do you agree with me?

Reply:
I think I need to read the code better to understand why we need both sorry!